lib/valueflow.h

```diff
@@ -244,7 +244,7 @@ namespace ValueFlow {
         /** The value bound  */
         enum class Bound { Upper, Lower, Point } bound;
 
-        /** int value */
+        /** int value (or sometimes bool value?) */
```

Comment:
As best I can tell, if the value is determined to be a bool then its value is stored in this member. I'm not personally a fan of that design but I've updated the documentation to reflect my understanding.

Reply:
well.. for me it does not feel surprising .. but if it can avoid a surprise for somebody else it's good to document it.

Reply:
It's not something I've given a lot of thought to so I'm certainly willing to accept that it's a well thought out design. If you can offer more explanation I'd be happy to expand on this documentation.

I was guessing it was probably an evolution of c static analysis which wouldn't require a distinction.